src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp

```diff
@@ -0,0 +1,111 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Mesh.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.tpp"
+#include "PointwiseFunctions/GeneralRelativity/Christoffel.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+template <size_t Dim, typename Frame>
+using deriv_tags = tmpl::list<
+    ::Tags::deriv<GeneralizedHarmonic::Tags::InitialGaugeH<Dim, Frame>,
+                  tmpl::size_t<Dim>, Frame>>;
+
+template <size_t Dim, typename Frame>
+using variables_tags =
+    tmpl::list<GeneralizedHarmonic::Tags::InitialGaugeH<Dim, Frame>>;
+}  // namespace
+
+namespace GeneralizedHarmonic {
+namespace gauges {
+namespace Actions {
+template <size_t Dim>
+std::tuple<tnsr::a<DataVector, Dim, Frame::Inertial>,
+           tnsr::ab<DataVector, Dim, Frame::Inertial>>
+InitializeDampedHarmonic<Dim>::impl(
+    const tnsr::aa<DataVector, Dim, Frame::Inertial>& spacetime_metric,
+    const tnsr::aa<DataVector, Dim, Frame::Inertial>& pi,
+    const tnsr::iaa<DataVector, Dim, Frame::Inertial>& phi,
+    const Mesh<Dim>& mesh,
+    const InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>&
+        inverse_jacobian) noexcept {
+  const auto spatial_metric = gr::spatial_metric(spacetime_metric);
```

Comment:
There are several temporaries being allocated in a row in this function, though I believe this is typically only run once during the evolution, so perhaps the allocation cost is not too high.